boa_engine/src/builtins/date/mod.rs

```diff
@@ -687,7 +788,16 @@ impl Date {
     ///
     /// [spec]: https://tc39.es/ecma262/#sec-date.prototype.gettime
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime
-    pub fn get_time(&self) -> f64 {
+    pub fn get_time(this: &JsValue, _args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
+        if let Some(t) = this_time_value(this, context)?.0 {
+            Ok(JsValue::new(t.timestamp_millis()))
+        } else {
+            Ok(JsValue::nan())
+        }
+    }
+
+    /// Utility: Internal `get_time()`
+    fn int_get_time(&self) -> f64 {
```

Comment:
Looking at the call-sites of this method, it should probably be named TimeClip according to the spec (https://tc39.es/ecma262/#sec-timeclip).

Reply:
`time_clip` already exits
```rust
#[inline]
pub fn time_clip(time: f64) -> Option<f64> {
    if time.abs() > 8.64e15 {
        None
    } else {
        Some(time)
    }
 }
```
`time_clip(f64)` and `int_get_time(self)` signatures are different and are being used at different places like [here](https://github.com/boa-dev/boa/blob/099154cb34c4f7fec6cf8d45f1f96b6797e10dd4/boa_engine/src/builtins/date/mod.rs#L321).

Let say I use `time_clip()` in `set_minutes()`, then I have to get milliseconds which is provided by `int_get_time()`.
So, Step 6 of `set_minutes` would look like this: `let u = Self::time_clip(t.int_get_time());`. And convert `Option<f64>` to `Result<JsValue, JsValue>`.

Looking at the [spec](https://tc39.es/ecma262/#sec-date.prototype.setminutes) its `LocalTime(t)`. 
On further look, steps doesn't match the spec for almost all the functions, is this intentional.

Thanks to this, I figured out the issue with `set_hours/minutes/etc.` step 6 expects `LocalTime` but `int_get_time` provides `UTC`.

Reply:
Alright, I think we should probably fix/refactor this. But then let's do that in another PR and keep the existing behaviour for this one.